spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -226,12 +205,31 @@ public void setCacheCapacity(int cacheCapacity) {
 		this.cacheCapacity = cacheCapacity;
 	}
 
+
+	/**
+	 * <p>
+	 * 	Because pub-sub does not work in some settings(RedisStaticMasterReplicaConfiguration), The default is {@link RedisLockType.SPIN_LOCK}
+	 * <ul>
+	 *    <li>{@link RedisLockType#SPIN_LOCK}: The lock is acquired by periodically(100ms) checking whether the lock can be acquired.</li>
+	 *    <li>{@link RedisLockType#PUB_SUB_LOCK}: The lock is accuired by redis pub-sub.</li>
+	 * 	</ul>
+	 * <p>
+	 * Set the type of unlockType, Select the lock method.
+	 *
+	 * @param redisLockType obtain RedisLockType
+	 * @since 5.5.13
+	 */
+	public void setRedisLockType(@NonNull RedisLockType redisLockType) {
```

Comment:
We assume that everything is not null, unless it is marked as `@Nullable`.
So, we typically don't use this annotation at all, unless we declare a `@NonNullApi` in the `package-info.java`.
Then we need to revise all the public API to honor nullability contract, so Kotlin projects won't fail.

Since there is only this `RedisLockRegistry` in this `org.springframework.integration.redis.util`, i won't mind if you address this concern in this PR as well.